Rubberduck.Core/UI/CodeExplorer/Commands/UpdateFromFileCommand.cs

```diff
@@ -30,194 +27,31 @@ public class UpdateFromFilesCommand : ImportCommand
-                return;
-            }
-
-            var documentFiles = moduleNames
-                .Select(kvp => kvp.Key)
-                .Where(filename => Path.GetExtension(filename) != null
-                              && ComponentTypeForExtension.TryGetValue(Path.GetExtension(filename), out var componentType)
-                              && componentType == ComponentType.Document)
-                .ToHashSet();
+        //Since we remove the components, we keep on the safe side.
+        protected override IEnumerable<string> AlwaysImportableExtensions => Enumerable.Empty<string>();
 
-            //We can only insert into existing documents.
-            if (!documentFiles.All(filename => modulesToRemoveBeforeImport.ContainsKey(filename)))
-            {
-                NotifyUserAboutAbortDueToNonExistingDocument(documentFiles, moduleNames, modulesToRemoveBeforeImport);
-                return;
-            }
-
-            //We must not remove document modules.
-            foreach (var filename in documentFiles)
-            {
-                modulesToRemoveBeforeImport.Remove(filename);
-            }
-
-            //We import the standalone code behind by replacing the code in an existing form.
-            //So, the form has to exist already.
-            if (!formFilesWithoutBinaries.All(filename => modulesToRemoveBeforeImport.ContainsKey(filename)))
-            {
-                NotifyUserAboutAbortDueToNonExistingUserForm(documentFiles, moduleNames, modulesToRemoveBeforeImport);
-                return;
-            }
-
-            foreach (var filename in formFilesWithoutBinaries)
-            {
-                modulesToRemoveBeforeImport.Remove(filename);
-            }
-
-            using (var components = targetProject.VBComponents)
-            {
-                foreach (var filename in filesToImport)
-                {
-                    if (modulesToRemoveBeforeImport.TryGetValue(filename, out var module))
-                    {
-                        var component = _projectsProvider.Component(module);
-                        components.Remove(component);
-                    }
-
-                    if(documentFiles.Contains(filename) || formBinaryModuleNames.Contains(filename))
-                    {
-                        //We have to dispose the return value.
-                        using (components.ImportSourceFile(filename)) { }
-                    }
-                    else
-                    {
-                        //We have to dispose the return value.
-                        using (components.Import(filename)) { }
-                    }
-                }
-            }
-        }
-
-        private Dictionary<string, string> ModuleNames(ICollection<string> filenames)
+        protected override bool ExistingModulesAreGenerallyOk(IDictionary<string, QualifiedModuleName> existingModules)
```

Comment:
could we rename to `ExistingModulesMayBeImported` or maybe `ExistingModulesAreEligible`?

Reply:
I do not think that any of those names would be right. It is more a `ExistingModulesAreNotInEligible`. That `True` is returned does not mean that they are OK, only that they are not deemed not OK.

Reply:
I see. The trouble is that `AreGenerallyOk` doesn't actually convey the reason why they are "OK". We also want to avoid the double negation as well. I think `MayBeImported` at least convey the possibility that they would be OK without necessarily saying it will import successfully. The other alternative I can think is to use words like `PreCheckPassed`, which would avoid suggesting that it's the final answer.